### PR TITLE
K8SPXC-321 feature: Restrict running less than 2 pods of proxySQL if unsafe flag is not set

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPXC-321](https://jira.percona.com/browse/K8SPXC-321) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPXC-321

Modified validation to restrict running less than 2 pods of proxySQL unless allowUnsafeConfigurations flag is set to true by changing proxySQL size to 2 if it's lower than that.

### CODE REVIEW

**Review by qJkee — DISMISSED** [review]

File: pkg/apis/pxc/v1/pxc_types.go [hunk 1]
```diff
@@ -215,6 +219,10 @@ func (cr *PerconaXtraDBCluster) Validate() error {
 		if err := c.ProxySQL.VolumeSpec.validate(); err != nil {
 			return errors.Wrap(err, "ProxySQL: validate volume spec")
 		}
+
+		if c.ProxySQL.Size < minSafeProxySQLSize && !c.AllowUnsafeConfig {
+			c.ProxySQL.Size = minSafeProxySQLSize
+		}
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_types.go:225]: I think `setSafeDefaults` function is better for setting new values. please excuse me for the confusion :(

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| haproxy | failed |
| upgrade-haproxy | passed |
| storage | failed |
| scaling-proxysql | failed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| upgrade-proxysql | passed |
| init-deploy | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| smart-update | failed |
| security-context | failed |
| limits | failed |
| monitoring-2-0 | passed |
| operator-self-healing | failed |
| affinity | failed |
| one-pod | passed |
| pitr | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | failed |
| demand-backup-encrypted-with-tls | passed |
| users | failed |
| scheduled-backup | failed |
| upgrade-consistency | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/f4713fc7da81f52c2fca0535cf7c3ec9496b1ef1
image: `perconalab/percona-xtradb-cluster-operator:PR-802-f4713fc7`

File: pkg/apis/pxc/v1/pxc_types.go (hunk 1 above)
Comment (pooknull) [reply pkg/apis/pxc/v1/pxc_types.go]: No problem. Fixed in 74953bd

File: pkg/apis/pxc/v1/pxc_types.go [hunk 2]
```diff
@@ -685,6 +689,10 @@ func setSafeDefaults(spec *PerconaXtraDBClusterSpec, log logr.Logger) {
 		loginfo("Cluster size will be changed from %d to %d due to safe config", spec.PXC.Size, spec.PXC.Size+1)
 		spec.PXC.Size++
 	}
+
+	if spec.ProxySQL.Size < minSafeProxySQLSize {
```
Comment (qJkee) [inline pkg/apis/pxc/v1/pxc_types.go:693]: We need to add check if `spec.ProxySQL` is not `nil` and enabled =)

**Review by qJkee — CHANGES_REQUESTED** [review]

File: pkg/apis/pxc/v1/pxc_types.go (hunk 2 above)
Comment (pooknull) [reply pkg/apis/pxc/v1/pxc_types.go]: Fixed in e6ab4ee9c0639938998154c8275c7e6564df8a08

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| haproxy | failed |
| big-data | passed |
| storage | failed |
| scaling-proxysql | failed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| upgrade-proxysql | passed |
| init-deploy | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| smart-update | failed |
| security-context | failed |
| limits | failed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | failed |
| one-pod | passed |
| pitr | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | failed |
| demand-backup-encrypted-with-tls | failed |
| users | failed |
| scheduled-backup | failed |
| upgrade-consistency | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/e6ab4ee9c0639938998154c8275c7e6564df8a08
image: `perconalab/percona-xtradb-cluster-operator:PR-802-e6ab4ee9`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| big-data | passed |
| haproxy | failed |
| storage | failed |
| upgrade-haproxy | passed |
| scaling-proxysql | failed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| init-deploy | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| smart-update | failed |
| security-context | failed |
| limits | failed |
| operator-self-healing | failed |
| monitoring-2-0 | failed |
| pitr | passed |
| demand-backup-encrypted-with-tls | failed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | failed |
| scheduled-backup | failed |
| upgrade-consistency | failed |
| users | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/74953bd3ca3d883ba486a9bce33fd95d0ee12ecd
image: `perconalab/percona-xtradb-cluster-operator:PR-802-74953bd3`

File: pkg/apis/pxc/v1/pxc_types.go (hunk 2 above)
Comment (qJkee) [reply pkg/apis/pxc/v1/pxc_types.go]: We also need to check if proxysql is enabled and log message according docs, if we set safe size

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/apis/pxc/v1/pxc_types.go [hunk 3]
```diff
@@ -685,6 +689,12 @@ func setSafeDefaults(spec *PerconaXtraDBClusterSpec, log logr.Logger) {
 		loginfo("Cluster size will be changed from %d to %d due to safe config", spec.PXC.Size, spec.PXC.Size+1)
 		spec.PXC.Size++
 	}
+
+	if spec.ProxySQL != nil {
+		if spec.ProxySQL.Size < minSafeProxySQLSize {
+			spec.ProxySQL.Size = minSafeProxySQLSize
+		}
+	}
```
[the remark this answers is not in the document]
Comment (pooknull) [reply pkg/apis/pxc/v1/pxc_types.go:697]: > according to the `As a user, I expect that operator will write a message to its log if it changes unsafe config to safe automatically` user story from the product requirements all changes should have a notification in the logs, please see 674-691 lines below

Shouldn't we add these notifications in task [K8SPXC-558](https://jira.percona.com/browse/K8SPXC-558)?

[the remark this answers is not in the document]
Comment (delgod) [reply pkg/apis/pxc/v1/pxc_types.go:697]: K8SPXC-558 was about old functionality, not about new checks.
K8SPXC-558 is fixed and closed.

**Review by delgod — APPROVED** [review]

**Review by qJkee — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| haproxy | failed |
| storage | failed |
| upgrade-haproxy | passed |
| recreate | passed |
| scaling | passed |
| scaling-proxysql | failed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| upgrade-proxysql | passed |
| limits | failed |
| security-context | failed |
| monitoring-2-0 | passed |
| operator-self-healing | failed |
| smart-update | failed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | failed |
| pitr | passed |
| users | failed |
| demand-backup-encrypted-with-tls | failed |
| scheduled-backup | failed |
| upgrade-consistency | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/1f6097938a57aab624829864cc20d929030bb1cb
image: `perconalab/percona-xtradb-cluster-operator:PR-802-1f609793`

**Comment by qJkee** [comment]
P.S it's also needed to change the compare files for tests

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| haproxy | passed |
| recreate | passed |
| scaling | passed |
| storage | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | failed |
| restore-to-encrypted-cluster | failed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| smart-update | failed |
| security-context | failed |
| demand-backup | failed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| pitr | passed |
| demand-backup-encrypted-with-tls | failed |
| upgrade-consistency | failed |
| tls-issue-self | passed |
| scheduled-backup | failed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/ace03f3d91f2cff8585a2b0cee57e8d613c59950
image: `perconalab/percona-xtradb-cluster-operator:PR-802-ace03f3d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | failed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| storage | passed |
| one-pod | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| upgrade-proxysql | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| smart-update | failed |
| demand-backup | passed |
| security-context | failed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | failed |
| upgrade-consistency | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/ace03f3d91f2cff8585a2b0cee57e8d613c59950
image: `perconalab/percona-xtradb-cluster-operator:PR-802-ace03f3d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| scaling | passed |
| scaling-proxysql | failed |
| big-data | passed |
| storage | passed |
| self-healing | passed |
| recreate | passed |
| self-healing-advanced | passed |
| restore-to-encrypted-cluster | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| haproxy | passed |
| scheduled-backup | failed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| upgrade-proxysql | passed |
| smart-update | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| security-context | failed |
| upgrade-consistency | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/ace03f3d91f2cff8585a2b0cee57e8d613c59950
image: `perconalab/percona-xtradb-cluster-operator:PR-802-ace03f3d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| upgrade-haproxy | passed |
| scaling | passed |
| storage | passed |
| scaling-proxysql | failed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | failed |
| upgrade-proxysql | passed |
| smart-update | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| security-context | failed |
| upgrade-consistency | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/ace03f3d91f2cff8585a2b0cee57e8d613c59950
image: `perconalab/percona-xtradb-cluster-operator:PR-802-ace03f3d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| recreate | passed |
| operator-self-healing | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| scaling | passed |
| demand-backup-encrypted-with-tls | passed |
| scaling-proxysql | failed |
| scheduled-backup | failed |
| upgrade-haproxy | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| upgrade-proxysql | passed |
| smart-update | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| security-context | failed |
| upgrade-consistency | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/ace03f3d91f2cff8585a2b0cee57e8d613c59950
image: `perconalab/percona-xtradb-cluster-operator:PR-802-ace03f3d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| big-data | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | failed |
| haproxy | passed |
| init-deploy | passed |
| scaling | passed |
| limits | passed |
| scaling-proxysql | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| upgrade-proxysql | passed |
| smart-update | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| security-context | failed |
| upgrade-consistency | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/ace03f3d91f2cff8585a2b0cee57e8d613c59950
image: `perconalab/percona-xtradb-cluster-operator:PR-802-ace03f3d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| haproxy | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| recreate | passed |
| big-data | passed |
| init-deploy | passed |
| self-healing | failed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| security-context | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | failed |
| affinity | passed |
| self-healing-advanced | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| users | passed |
| tls-issue-self | passed |
| pitr | passed |
| tls-issue-cert-manager | failed |
| demand-backup-encrypted-with-tls | passed |
| upgrade-consistency | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| scheduled-backup | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/647eb0e6c5fc6f8adcb6c91e539eaa59533bc13e
image: `perconalab/percona-xtradb-cluster-operator:PR-802-647eb0e6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| haproxy | passed |
| recreate | passed |
| scaling | passed |
| storage | passed |
| big-data | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | failed |
| security-context | passed |
| upgrade-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| monitoring-2-0 | passed |
| smart-update | passed |
| affinity | passed |
| demand-backup | passed |
| one-pod | passed |
| self-healing-advanced | passed |
| pitr | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | passed |
| proxysql-sidecar-res-limits | failed |
| scheduled-backup | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/dfd601fa58eed4854a275305924765d525121c3a
image: `perconalab/percona-xtradb-cluster-operator:PR-802-dfd601fa`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| haproxy | passed |
| recreate | passed |
| upgrade-haproxy | passed |
| big-data | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | passed |
| users | failed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | failed |
| tls-issue-self | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/b61876ae79ba13f49afd89f9e886606bd9fb3dac
image: `perconalab/percona-xtradb-cluster-operator:PR-802-b61876ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| big-data | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/b61876ae79ba13f49afd89f9e886606bd9fb3dac
image: `perconalab/percona-xtradb-cluster-operator:PR-802-b61876ae`

File: e2e-tests/scaling-proxysql/run [hunk 4]
```diff
@@ -52,15 +52,11 @@ apply_config $test_dir/conf/$cluster.yml
...
 wait_for_delete pod/$cluster-proxysql-2
-wait_for_delete pod/$cluster-proxysql-1
 
 desc 'check if PVC not deleted'
-compare_kubectl pvc/proxydata-$cluster-proxysql-1
 compare_kubectl pvc/proxydata-$cluster-proxysql-2
 
 desc 'check if Pod deleted from ProxySQL'
-run_mysql_local 'SELECT hostname FROM runtime_proxysql_servers;' "-h127.0.0.1 -P6032 -uproxyadmin -padmin_password" "$cluster-proxysql-0" \
-    | ( grep $pod1 && exit 1 || :)
```
Comment (delgod) [inline e2e-tests/scaling-proxysql/run:63]: why these lines are deleted?

File: e2e-tests/smart-update/run [hunk 5]
```diff
@@ -10,7 +10,8 @@ API='pxc.percona.com/v9-9-9'
 TARGET_IMAGE_PXC=${IMAGE_PXC}
 CLUSTER="smart-update"
 CLUSTER_SIZE=3
-PROXY_SIZE=1
+PROXYSQL_SIZE=2
+HAPROXY_SIZE=1
```
Comment (delgod) [inline e2e-tests/smart-update/run:14]: why we need two variables? minimal size is `2` for both

**Review by delgod — CHANGES_REQUESTED** [review]

File: e2e-tests/scaling-proxysql/run (hunk 4 above)
Comment (ollevche) [reply e2e-tests/scaling-proxysql/run:63]: because there is only one pod deleted due to safe config

File: e2e-tests/smart-update/run (hunk 5 above)
Comment (ollevche) [reply e2e-tests/smart-update/run:14]: it's not implemented in this PR, but we can add [K8SPXC-556](https://jira.percona.com/browse/K8SPXC-556) here

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/b61876ae79ba13f49afd89f9e886606bd9fb3dac
image: `perconalab/percona-xtradb-cluster-operator:PR-802-b61876ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| scaling | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| big-data | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/b61876ae79ba13f49afd89f9e886606bd9fb3dac
image: `perconalab/percona-xtradb-cluster-operator:PR-802-b61876ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| upgrade-haproxy | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/b61876ae79ba13f49afd89f9e886606bd9fb3dac
image: `perconalab/percona-xtradb-cluster-operator:PR-802-b61876ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | passed |
| scaling | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling-proxysql | passed |
| affinity | passed |
| security-context | passed |
| one-pod | passed |
| auto-tuning | passed |
| upgrade-haproxy | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | passed |
| big-data | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/b61876ae79ba13f49afd89f9e886606bd9fb3dac
image: `perconalab/percona-xtradb-cluster-operator:PR-802-b61876ae`

**Review by qJkee — APPROVED** [review]

**Review by delgod — DISMISSED** [review]

**Review by qJkee — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| storage | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| affinity | passed |
| pitr | passed |
| one-pod | passed |
| auto-tuning | passed |
| upgrade-consistency | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/6d7d8d8664fee32784a8570bc18dea57873b8442
image: `perconalab/percona-xtradb-cluster-operator:PR-802-6d7d8d86`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| big-data | passed |
| operator-self-healing | passed |
| haproxy | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | failed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| scheduled-backup | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/6d7d8d8664fee32784a8570bc18dea57873b8442
image: `perconalab/percona-xtradb-cluster-operator:PR-802-6d7d8d86`

**Comment by qJkee** [comment]
@ollevche 
monitoring-2-0 test fails on diff. Need to update compare files

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| recreate | passed |
| storage | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| haproxy | passed |
| self-healing | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | passed |
| users | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| scheduled-backup | failed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/2c303781cde279d2a45e0d07411bdc97a4bc82b4
image: `perconalab/percona-xtradb-cluster-operator:PR-802-2c303781`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| upgrade-haproxy | passed |
| big-data | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/2c303781cde279d2a45e0d07411bdc97a4bc82b4
image: `perconalab/percona-xtradb-cluster-operator:PR-802-2c303781`

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| upgrade-haproxy | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scaling | passed |
| users | passed |
| scaling-proxysql | passed |
| storage | passed |
| tls-issue-self | passed |
| security-context | passed |
| self-healing | passed |
| tls-issue-cert-manager | passed |
| self-healing-advanced | passed |
| tls-issue-cert-manager-ref | passed |
| operator-self-healing | passed |
| validation-hook | passed |
| big-data | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/2c303781cde279d2a45e0d07411bdc97a4bc82b4
image: `perconalab/percona-xtradb-cluster-operator:PR-802-2c303781`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| storage | passed |
| auto-tuning | passed |
| self-healing | passed |
| proxysql-sidecar-res-limits | passed |
| self-healing-advanced | passed |
| users | passed |
| operator-self-healing | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| scaling | passed |
| tls-issue-cert-manager-ref | passed |
| scaling-proxysql | passed |
| validation-hook | passed |
| security-context | passed |
| big-data | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/802/commits/2c303781cde279d2a45e0d07411bdc97a4bc82b4
image: `perconalab/percona-xtradb-cluster-operator:PR-802-2c303781`

**Pull request merged**